app-backend/api/src/main/scala/user/Routes.scala

```diff
@@ -271,4 +280,61 @@ trait UserRoutes
       }
     }
   }
+
+  def createUserBulk: Route = authenticate { user =>
+    authorizeScope(
+      ScopedAction(Domain.Organizations, Action.AddUser, None),
+      user
+    ) {
+      entity(as[UserBulkCreate]) { userBulkCreate =>
+        complete {
+          Auth0Service.getManagementBearerToken flatMap { managementToken =>
+            val names = PseudoUsernameService.createPseudoNames(
+              userBulkCreate.count,
+              userBulkCreate.peudoUserNameType
+            )
+            for {
+              userNames <- names traverse { name =>
+                for {
+                  auth0User <- Auth0Service.createERUser(name, managementToken)
+                  newUser <- (auth0User.user_id traverse { userId =>
+                    for {
```

Comment:
it's a little arbitrary but the third nested for comprehension is really making me cross my eyes -- can you pull out the create user/create default roles/copy campaign flow to a function in `UserDao` that indicates the anonymized user creation flow?